library/src/test/java/io/constructor/core/ConstructorioBrowseItemsTest.kt

```diff
@@ -0,0 +1,270 @@
+    private val preferencesHelper = mockk<PreferencesHelper>()
+    private val configMemoryHolder = mockk<ConfigMemoryHolder>()
+    private val ids = listOf("dai_pid_2003438", "dai_pid_2003597")
+
+    @Before
+    fun setup() {
+        mockServer = MockWebServer()
+        mockServer.start()
+
+        every { ctx.applicationContext } returns ctx
+
+        every { preferencesHelper.apiKey } returns "silver-key"
+        every { preferencesHelper.id } returns "guapo-the-guid"
+        every { preferencesHelper.serviceUrl } returns mockServer.hostName
+        every { preferencesHelper.port } returns mockServer.port
+        every { preferencesHelper.scheme } returns "http"
+        every { preferencesHelper.defaultItemSection } returns "Products"
+        every { preferencesHelper.getSessionId(any(), any()) } returns 92
+
+        every { configMemoryHolder.autocompleteResultCount } returns null
+        every { configMemoryHolder.userId } returns "player-two"
+        every { configMemoryHolder.testCellParams } returns emptyList()
+        every { configMemoryHolder.segments } returns emptyList()
+
+        val config = ConstructorIoConfig("dummyKey")
+        val dataManager = createTestDataManager(preferencesHelper, configMemoryHolder)
+
+        constructorIo.testInit(ctx, config, dataManager, preferencesHelper, configMemoryHolder)
+    }
+
+    @Test
+    fun getBrowseItemsResults() {
+        val mockResponse = MockResponse().setResponseCode(200)
+            .setBody(TestDataLoader.loadAsString("browse_response.json"))
+        mockServer.enqueue(mockResponse)
+        val browseItemsRequest = BrowseItemsRequest.Builder(ids)
+            .build()
+        val observer = constructorIo.getBrowseItemsResults(browseItemsRequest).test()
+        observer.assertComplete()
+        observer.assertNoErrors()
+
+        val browseResponse = observer.values()[0].get()
+        assertEquals(browseResponse?.response?.results?.size, 5)
+        assertEquals(browseResponse?.response?.results!![0].value, "Crystal Geyser Natural Alpine Spring Water - 1 Gallon")
+        assertEquals(browseResponse.response?.results!![0].data.id, "108200440")
+        assertEquals(browseResponse.response?.results!![0].data.imageUrl, "https://d17bbgoo3npfov.cloudfront.net/images/farmstand-108200440.png")
+        assertEquals(browseResponse.response?.results!![0].data.metadata?.get("price"), 1.25)
+        assertEquals(browseResponse.response?.facets!!.size, 3)
+        assertEquals(browseResponse.response?.facets!![0].displayName, "Brand")
+        assertEquals(browseResponse.response?.facets!![0].type, "multiple")
+        assertEquals(browseResponse.response?.groups!!.size, 1)
+        assertEquals(browseResponse.response?.resultCount, 367)
+
+        val request = mockServer.takeRequest()
+        val path =
+            "/browse/items?ids=dai_pid_2003438&ids=dai_pid_2003597&key=silver-key&i=guapo-the-guid&ui=player-two&s=92&c=cioand-2.19.1&_dt="
```

Comment:
Looks like all of the tests in this file are failing. I think it's expecting the `c` param to be `cioand-2.19.3`